scss/_patterns_notifications.scss

```diff
@@ -1,62 +1,146 @@
+$notification-text-margin-bottom: 0.6rem;
 
 // Notification style patterns
 @mixin vf-p-notification {
   // The mixin for basic notification styling
   %vf-notification {
     @extend %vf-is-bordered;
 
+    background-position: $sph-inner $notification-icon-vert-offset;
+    background-repeat: no-repeat;
+    background-size: map-get($icon-sizes, default);
     border-radius: 0 $border-radius $border-radius 0;
-    display: flex;
     margin-bottom: $spv-outer--scaleable;
-    overflow: hidden;
-    padding: 0;
+    margin-top: -1px;
+    padding-left: $notification-content-icon-space;
+    position: relative;
 
     &.is-borderless {
+      background-color: transparent;
+      background-position-x: 0;
       border-color: transparent;
-
-      .p-notification__content {
-        background-color: transparent;
-        background-position: 0 0.5 * (map-get($line-heights, default-text) - map-get($icon-sizes, default));
-        padding: 0 0 0 2 * $sph-inner;
-      }
+      margin-top: 0;
+      padding: 0 0 0 2 * $sph-inner;
 
       &::before {
         display: none;
       }
     }
 
     &.is-inline {
-      .p-notification__title {
-        display: inline;
+      .p-notification__content {
+        margin-bottom: $notification-text-margin-bottom;
+        padding-top: map-get($nudges, nudge--p) + map-get($browser-rounding-compensations, h5);
       }
 
+      .p-notification__title,
       .p-notification__message {
         display: inline;
+        margin: 0;
+        padding: 0;
+      }
+    }
+
+    &.is-modal {
+      box-shadow: $box-shadow;
+
+      .p-notification__actions:only-child {
+        margin-left: unset;
+      }
+
+      &::before {
+        display: none;
+      }
+    }
+
+    &.is-raised {
+      box-shadow: 0 0 2rem 0 rgba($color-x-dark, $shadow-opacity);
+    }
+
+    &.is-unread {
+      .p-notification__title {
+        position: relative;
+
+        &::after {
+          @include vf-icon-status-small($color-link);
+          @include vf-icon-size($default-icon-size);
+
+          content: '';
+          position: absolute;
+          top: $notification-icon-vert-offset + map-get($nudges, nudge--small);
+        }
       }
     }
   }
 
   .p-notification__content {
-    background-position: $sph-inner $notification-content-vert-padding + $notification-icon-vert-offset;
-    background-repeat: no-repeat;
-    background-size: map-get($icon-sizes, default);
-    margin-bottom: 0;
-    margin-top: -1px;
-    padding: $notification-content-vert-padding $sph-inner + $spv-inner--small $notification-content-vert-padding $notification-content-icon-space;
+    padding-right: 2 * $sph-inner;
   }
 
   .p-notification__title {
-    @extend %bold;
+    @extend %vf-heading-5;
 
-    display: block;
+    margin-bottom: 0.1rem;
   }
 
   .p-notification__message {
-    display: block;
+    @extend %default-text;
+
+    margin-bottom: $notification-text-margin-bottom;
+  }
+
+  .p-notification__close {
```

Comment:
The design is not clear on that, but I think close button should be a standard button (base button style) - so icon will be visible and hover will make button shape appear.

This will help aligning it with contextual menu (when it's used), but also means the position of it will need to be adjusted so it moves a bit into padding.

Reply:
Yeah this is a bit finicky - we're trying to make the icon line up with the title text and the contextual menu icon *and* not be too close to the borders but it's kinda hard to satisfy all three conditions.